common/static/common/js/discussion/discussion_router.js

```diff
@@ -127,17 +135,14 @@
             };
 
             DiscussionRouter.prototype.navigateToThread = function(thread_id) {
-                var thread;
+                var thread, targetThreadRoute;
                 thread = this.discussion.get(thread_id);
-                return this.navigate("" + (thread.get("commentable_id")) + "/threads/" + thread_id, {
-                    trigger: true
-                });
+                targetThreadRoute = getSingleThreadRoute(thread.get("commentable_id"), thread_id);
+                this.navigate(targetThreadRoute, { trigger: true});
```

Comment:
@bradenmacdonald yes:
* in [Backbone docs](http://backbonejs.org/#Router-navigate) navigate return value is never used
* it's just coffeescript always returning the result of last line of code